transports/quic/src/transport.rs

```diff
@@ -49,12 +49,23 @@ use std::{
 };
 
 /// Implementation of the [`Transport`] trait for QUIC.
+///
+/// Per default only QUIC Version 1 (RFC 9000) is supported. In the [`Multiaddr`] this maps to
+/// [`libp2p_core::multiaddr::Protocol::QuicV1`].
+/// The [`libp2p_core::multiaddr::Protocol::Quic`] codepoint is interpreted as QUIC version
+/// draft-29 and only supported if [`Config::support_draft_29`] is set to `true`.
+/// Note that in that case servers support both version an all QUIC listening addresses.
+///
+/// Version draft-29 should only be used to connect to nodes from other libp2p implementations
+/// that do not support `QuicV1` yet. Support for it will be removed long-term.
```

Comment:
How about linking to https://github.com/multiformats/multiaddr/issues/145 here?

Reply:
Done in fa39fc651ff49f1a173ced0c2c7cc05a57e27242.